external-crates/move/crates/move-cli/src/base/build.rs

```diff
@@ -23,7 +23,29 @@ impl Build {
             return Ok(());
         }
 
-        config.compile_package(&rerooted_path, &mut std::io::stdout())?;
+        let result = config
+            .clone()
+            .cli_compile_package(&rerooted_path, &mut std::io::stdout());
+        if let Err(err) = &result {
+            let err_msg = err.downcast_ref::<&str>();
+            if let Some(str) = err_msg {
+                if matches!(str, &NEEDS_MIGRATION) {
```

Comment:
We don't care about the diagnostic code, because we only generate migration diagnostics when we are in migration mode. We could avoid this check here by passing `stdin` to `cli_compile_package` and do the check on that side, though. Let me know if you'd like me to make that change.

Reply:
Might be cleaner to add a `cli_compile_package_with_driver` to mirror the `compile_with_driver` API of the package system. This way we could compile, and then check the diags before reporting. Thoughts?